sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -561,11 +608,53 @@ public virtual int GetBufferedEventCount(string partitionId)
         ///   <see cref="SendEventBatchFailedAsync" /> handlers will be validated and can no longer be changed.
         /// </remarks>
         ///
-        public virtual Task<int> EnqueueEventAsync(EventData eventData,
-                                                   EnqueueEventOptions options,
-                                                   CancellationToken cancellationToken = default)
+        public virtual async Task<int> EnqueueEventAsync(EventData eventData,
+                                                         EnqueueEventOptions options,
+                                                         CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            // Validate
+
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
+
+            // Log
```

Comment:
Is the log statement missing?

Reply:
It's a stub.   I'll annotate this better and update.